core/src/main/java/org/apache/iceberg/ScanSummary.java

```diff
@@ -305,7 +305,7 @@ public String toString() {
   private static class TopN<K, V> {
     private final int maxSize;
     private final boolean throwIfLimited;
-    private final SortedMap<K, V> map;
+    private final NavigableMap<K, V> map;
```

Comment:
Curious to what the issue that this change fixes? I'm asking cause I'm not really that knowledgeable when it comes to these java collections in particular, all I know is that SortedMap is a superinterface of NavigableMap, and since I'm generally guided by using the least specific interface "principle" I was wondering why that wouldn't apply in this case as well.

Reply:
Great question. This one triggered me:
```
/Users/fokkodriesprong/Desktop/incubator-iceberg/core/src/main/java/org/apache/iceberg/ScanSummary.java:326: warning: [JdkObsolete] SortedMap was replaced by NavigableMap in Java 6.
      map.put(key, updateFunc.apply(map.get(key)));
             ^
    (see https://errorprone.info/bugpattern/JdkObsolete)
```
The SortedMap has been superseded by the NavigableMap. According to the author, there are some issues with the sorted set API: https://www.youtube.com/watch?v=aAb7hSCtvGw#t=1258 I agree that in this case, it doesn't really matter.